strabo/static/js/popup.js

```diff
@@ -139,21 +139,55 @@ function make_photoswipe(pic_index){
 
     gallery.init();
 }
-function set_flickety_img_title(){
+function linear_text_width(text){
+    /*Returns width of text (in pixels) if it never wraps, assuming a font size of a
+    single pixel.*/
+
+    //adds a really wide temporary div to screen to allow for measurement of long text
+    var $meas_space = $('<div class="img-description text-measure"></div>')
+    $("body").append($meas_space);
+    //adds text to div for measurement.
+    var $span = $("<p>"+text+"</p>");
+    $meas_space.append($span);
+
+    var width = $span.width();
+
+    $meas_space.remove()
+    //.text-measure css class is 10px because really small text renders differently.
+    //and so this is more accurate.
+    var measured_width = 10.0;
+    //gives room for 10% error in text size calculation
+    var measured_error = 1.1;
+    return (width*measured_error) / measured_width;
+}
+function get_img_desc_font_size(imgdescr){
+    var $container = $("#img-text-section")
+
+    var text_width_max = $container.width() / linear_text_width(imgdescr);
+    //makes sure bottom of text is not cut off
+    var text_height_max = $container.height() / 1.25;
+
+    return Math.min(text_width_max, text_height_max);
+}
+function set_flickity_img_title(){
     flkty.on( 'cellSelect', function() {
         var img = imgs[flkty.selectedIndex];
 
-        $("#img_description").text(img.description);
+        var img_desc = document.getElementById("img-descr");
```

Comment:
ah, this is what I meant about the inconsistency in "description." img_desc vs img-descr